### ISSUE TITLE
Implement instanced rooms/room groups.

### ISSUE THREAD

**Issue by bbuck** [bbuck/dragon-mud#17]
Just like in popular MMOs where certain dungeons or areas, upon entrance, spawn the player or groups own instance of that area. Allowing two groups to run the same dungeon at the same time without interfering with one another or causing problems.

This would allow exits to be marked as instance entrances which would spawn the room or a group of rooms for that player and/or group. Another usage of this would be to segment highly trafficked areas. Instead of dealing with 300 people in one room, the room could be segmented by instances of say, 50 players or so (not that this is the desired effect) and so forth.

To sum up reasons for instancing:
- Ensure each player/group can experience certain dungeon content without having interference from other players
- Allow segmenting highly populated areas to prevent unnecessary spam in chats
- Allow for special one off versions of other rooms (conditional instancing) for certain quests.

**Comment by bbuck** [comment]
Probably best to implement this in such a way that rooms are instanced as they're entered. Once I (or someone) comes around to implementing this it would be great to compare to find out at what size of an instance copying an entire instance over would become inefficient (making the player noticeably wait). Or if copying each room individually if they're instanced in such a way they share an instance ID of some kind.
